### PR TITLE
Zaphod/fix join new game

### PR DESCRIPTION
**Pull request by zaphodbeblebrox**
Selecting a player that has votes will load the SetCount modal with that value instead of zero.
image

Player Tab in player view will now display who is being executed and number of votes needed to execute. Also fixed # votes needed to execute when there is a tie.
image

^ and DesktopView (spectator view) shared the same code to complete this. Code extracted to new component ExecuteInfo.tsx
image

### CODE REVIEW

File: client/src/GamemasterDesktop/DesktopView.tsx [hunk 1]
```diff
@@ -125,6 +107,9 @@ function Grimoire({ isPlayerView = true }: GrimoireProps) {
                   <SetCountModal
                     title="Votes to Execute:"
                     onSet={(votes: number) => setVotesToExecute(player, votes)}
+                    defaultValue={
+                      game.onTheBlock[player] ? game.onTheBlock[player] : 0
```
Comment (vermentor) [inline client/src/GamemasterDesktop/DesktopView.tsx:111]: You should be able to simplify this as `game.onTheBlock[player] ?? 0` I believe.

File: client/src/PlayerLanding/PlayerInGame.tsx [hunk 2]
```diff
@@ -73,6 +74,9 @@ export function PlayerInGame() {
     return nightOrder;
   }, [script, game.playersToRoles]);
 
+  // TODO: Delete this
+  console.log(game);
```
Comment (vermentor) [inline client/src/PlayerLanding/PlayerInGame.tsx:78]: Todo

File: client/src/shared/ExecutionInfo.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { UnifiedGame } from "@hidden-identity/shared";
+import { usePlayerOrder } from "./PlayerListOrder";
+import { Text } from "@radix-ui/themes";
+
+interface ExecutionInfoProps {
+  game: UnifiedGame;
+  firstSeat: string | null;
```
Comment (vermentor) [inline client/src/shared/ExecutionInfo.tsx:7]: Neither of these props should be necessary to this component.
`game` can be acquired via `useDefiniteGame` which holds game in a global context.
`players` can be read from game as `game.playerList` or some similarly named property.

File: client/src/shared/GameHeader.tsx [hunk 4]
```diff
@@ -40,7 +41,9 @@ export function GameHeader() {
         <Callout.Root>
           <Callout.Text>
             <Link asChild>
-              <ReactRouterLink to={`/${game.nextGameId}`}>
+              <ReactRouterLink
+                to={urlFromOrigin("game", { gameId: game.nextGameId })}
+              >
```
Comment (vermentor) [inline client/src/shared/GameHeader.tsx:46]: Out of scope but I'll allow it.

Comment (zaphodbeblebrox) [reply client/src/shared/GameHeader.tsx:46]: weird but should already be in main

File: client/src/shared/GameHeader.tsx [hunk 5]
```diff
@@ -55,7 +58,6 @@ interface QRCodeDialogProps {
   gameId: string | null;
 }
 function QRCodeDialog({ gameId }: QRCodeDialogProps) {
-  // TODO: update to use url.ts
```
Comment (Alphy11) [inline client/src/shared/GameHeader.tsx:58]: You removed this todo but didn't fix the issue

File: client/src/shared/ExecutionInfo.tsx [hunk 6]
```diff
@@ -0,0 +1,40 @@
+import { Text } from "@radix-ui/themes";
+import { useDefiniteGame } from "../store/GameContext";
+
+export function ExecutionInfo() {
+  const { game } = useDefiniteGame();
+  // const players = usePlayerOrder("seat order", firstSeat);
```
Comment (Alphy11) [inline client/src/shared/ExecutionInfo.tsx:6]: Remove these if they aren't needed

File: client/src/PlayerLanding/PlayerInGame.tsx [hunk 7]
```diff
@@ -195,22 +182,40 @@ export function PlayerInGame() {
...
+                    as="div"
+                    className="capitalize"
+                  >
+                    Votes: {game.onTheBlock[player]}
+                  </Text>
                 )}
                 <SetCountModal
                   title="Set votes to execute:"
                   onSet={(votes: number) => setVotesToExecute(player, votes)}
+                  defaultValue={game.onTheBlock[player] ?? 0}
```
Comment (Alphy11) [inline client/src/PlayerLanding/PlayerInGame.tsx:218]: Let's get rid of these props and hook the modal up to the game directly.

**Comment by zaphodbeblebrox** [comment]
image

**Review by Alphy11 — APPROVED** [review]

**Pull request merged**